### PR TITLE
Warn about non applied pagination config

### PR DESCRIPTION
**Pull request by dereuromark**
Refs https://github.com/cakephp/docs/pull/7968

    protected array $paginate = [
        'finder' => 'foo',

would be working for

    $banks = $this->paginate('Banks');

but not for

    $query = $this->Banks->find();
    $banks = $this->paginate($query);

and that without any hint/notice.
Only debugging it very deeply into the paginator code you find out why.

So if you refactor your code to modern 5.1 (baked) code you can lose this filter without any clue that it happened.

I suggest the following warning or error or alike:

> Custom finder `foo` from pagination config cannot be applied to a custom build query object - /shared/httpd/app/vendor/cakephp/cakephp/src/Datasource/Paging/NumericPaginator.php, line: 223

Maybe we can rephrase it to `to a manually built SelectQuery object` ?

### CODE REVIEW

**Comment by ADmad** [comment]
Finders can be applied to queries too, so how about we do that instead in 5.2? Changing this behavior could cause unexpected results, so safer to do in a new minor.

**Comment by markstory** [comment]
> Finders can be applied to queries too, so how about we do that instead in 5.2? 

So no warning for now? Or we merge the warning now and then change the behavior for 5.2?

**Comment by dereuromark** [comment]
I think we should warn now and can then remove the warning once they can actually work and be applied (pending 5.2 PR).
Right now any 5.1 user would silently have a no-op.

**Review by ADmad — APPROVED** [review]

**Comment by dereuromark** [comment]
I would still rephrase it to e.g.

    ... to a manually built SelectQuery object

or alike

File: src/Datasource/Paging/NumericPaginator.php [hunk 1]
```diff
@@ -274,6 +274,11 @@ protected function getQuery(RepositoryInterface $object, ?QueryInterface $query,
             }
 
             $query = $object->find($type, ...$args);
+        } elseif (!empty($options['finder'])) {
+            triggerWarning(sprintf(
+                'Custom finder `%s` from pagination config cannot be applied to a custom build query object',
```
Comment (ADmad) [inline src/Datasource/Paging/NumericPaginator.php]:
```suggestion
                'Finder option (`%s`) from pagination config is not applied when a `SelectQuery` instance is passed to `paginate()`',
```

**Comment by dereuromark** [comment]
This

> Line exceeds 120 characters; contains 133 characters

is so terribly outdated, at least 140 should be FINE IMO

**Comment by ADmad** [comment]
Since the `finder` config defaults to `all`, this patch currently shows deprecation warning for *every* call to `paginate()` with a query. It needs to be skipped when `finder` is `all`.
